Mage.Sets/src/mage/cards/t/TaintedSpecter.java

```diff
@@ -0,0 +1,114 @@
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.TargetPlayer;
+import mage.target.common.TargetCardInHand;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ * @author Cguy7777
+ */
+public final class TaintedSpecter extends CardImpl {
+
+    public TaintedSpecter(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{B}");
+
+        this.subtype.add(SubType.SPECTER);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // {1}{B}{B}, {tap}: Target player discards a card unless they put a card from their hand on top of their library.
+        // If that player discards a card this way, Tainted Specter deals 1 damage to each creature and each player.
+        // Activate this ability only any time you could cast a sorcery.
+        Ability ability = new ActivateAsSorceryActivatedAbility(
+                Zone.BATTLEFIELD, new TaintedSpecterEffect(), new ManaCostsImpl<>("{1}{B}{B}"));
+        ability.addCost(new TapSourceCost());
+        ability.addTarget(new TargetPlayer());
+        this.addAbility(ability);
+    }
+
+    private TaintedSpecter(final TaintedSpecter card) {
+        super(card);
+    }
+
+    @Override
+    public TaintedSpecter copy() {
+        return new TaintedSpecter(this);
+    }
+}
+
+class TaintedSpecterEffect extends OneShotEffect {
+
+    public TaintedSpecterEffect() {
+        super(Outcome.Detriment);
+        staticText = "Target player discards a card unless they put a card from their hand on top of their library. "
+                + "If that player discards a card this way, {this} deals 1 damage to each creature and each player";
+    }
+
+    private TaintedSpecterEffect(final TaintedSpecterEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player targetPlayer = game.getPlayer(this.getTargetPointer().getFirst(game, source));
+        if (targetPlayer == null) {
+            return false;
+        }
+
+        if (targetPlayer.getHand().isEmpty()) {
+            return true;
+        }
+
+        final String message = "Discard a card, or put a card from your hand on top of your library?";
+        if (targetPlayer.chooseUse(outcome, message, null, "Discard a card", "Put a card from your hand on top", source, game)) {
+            // Discard a card
+            Effect discardEffect = new DiscardTargetEffect(1);
+            discardEffect.setTargetPointer(new FixedTarget(targetPlayer.getId()));
+
+            if (discardEffect.apply(game, source)) {
+                new DamageEverythingEffect(1).apply(game, source);
```

Comment:
(I have to double check this class every time because it is poorly named- functionality is correct though)